client/client-multi/client-c/src/kaa_logging.c

```diff
@@ -220,37 +220,51 @@ kaa_error_t kaa_logging_request_serialize(kaa_log_collector_t *self, kaa_platfor
         return KAA_ERR_BAD_STATE;
     ++self->log_bucket_id;
 
-    *((uint16_t *) writer->current) = KAA_HTONS(self->log_bucket_id);
-    writer->current += sizeof(uint16_t);
-    char *records_count_p = writer->current; // pointer to the records count. Will be filled later.
-    writer->current += sizeof(uint16_t);
+    *((uint16_t *) tmp_writer.current) = KAA_HTONS(self->log_bucket_id);
+    tmp_writer.current += sizeof(uint16_t);
+    char *records_count_p = tmp_writer.current; // Pointer to the records count. Will be filled in later.
+    tmp_writer.current += sizeof(uint16_t);
 
-    ssize_t remaining_size = self->log_properties.max_log_bucket_size;
-    KAA_LOG_TRACE(self->logger, KAA_ERR_NONE, "Extracting log records... (Bucket size is %zu)", remaining_size);
+    ssize_t remaining_size = self->max_log_bucket_size < (tmp_writer.end - tmp_writer.current)
```

Comment:
Why remaining size isn't set to the max bucket size when the size of the unused space in a buffer matches to it?